include/pybind11/detail/internals.h

```diff
@@ -43,6 +43,9 @@ using ExceptionTranslator = void (*)(std::exception_ptr);
 
 PYBIND11_NAMESPACE_BEGIN(detail)
 
+static constexpr const char *internals_function_record_capsule_name
```

Comment:
What does static do in this context? Add a comment?

Reply:
That was my suggestion: I'm never entirely sure where and why `static` is needed.
But clang-tidy was happy.
I don't think it needs a comment, the intent is clear. If a tool complains in the future, we'll fix it.

Reply:
@rwgk requested static there, not sure why? I also don't quite understand the reason for the static when it is already constexpr?

Reply:
In class context it's definitely needed, e.g:

```
rwgk.c.googlers.com:~/forked/pybind11/include $ grep -r 'static constexpr' | wc -l
117
```

`namespace`, I'm not sure.

But generally I try not to waste too many of the few neurons that I have on clearly inconsequential details of an extremely quirky language. If a tool tells me to do it differently I comply. Our tooling is almost insanely thorough. I stop thinking there.

Reply:
> I'm never entirely sure where and why static is needed

You need static whenever you ODR use a constexpr variable that is in a header file to avoid linking errors. 

One of the annoying traps in the C++ language. This case probably dosen't need static because we don't ODR use it.